#! /bin/sh -e
## 62_new_list_bad_pending_requests.dpatch by richm@oldelvet.org.uk
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: update is called from the Debian maintainer scripts and prints
## DP: non-appropriate output.  Change that.

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch ${2:+-d $2}}"

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $patch_opts -p1 < $0;;
    -unpatch) patch $patch_opts -p1 -R < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0
@DPATCH@
--- mailman/Mailman/ListAdmin.py.orig   2004-09-26 22:11:59.412607418 +0100
+++ mailman/Mailman/ListAdmin.py        2004-09-26 22:19:43.623697828 +0100
@@ -122,8 +122,11 @@

     def NumRequestsPending(self):
         self.__opendb()
-        # Subtrace one for the version pseudo-entry
-        return len(self.__db) - 1
+       if self.__db.has_key('version'):
+           # Subtract one for the version pseudo-entry
+           return len(self.__db) - 1
+       else:
+           return len(self.__db)

     def __getmsgids(self, rtype):
         self.__opendb()
